ui/webapp.py

```diff
@@ -180,7 +186,7 @@ def reset_results(*args):
                 st.write("**Relevance:** ", result["relevance"], "**Source:** ", result["source"])
 
             else:
-                st.warning("🤔 &nbsp;&nbsp; Haystack found no good answer to your question. Try to formulate it differently!")
+                st.info("🤔 &nbsp;&nbsp; Haystack found no good answer to your question. Try to formulate it differently!")
```

Comment:
How about: "Haystack found no good answer. If you think the answer is really somewhere in the documents, try to formulate your question differently!"

We should probably filter out this "no answer" block in the results if it is not the "top 1". What do you think?

Reply:
Good question. I find them marginally useful even when they're not the first position because they're a signal that whatever answer is below them is more of a guess than an actual answer. On the other hand, I could just filter out all answers that come out below a no-answer; even though sometimes the right answer might be below it.

Right now I lean towards leaving it as it is, but happy to apply the filter if you prefer so. Maybe @julian-risch has an opinion here too :slightly_smiling_face:

Reply:
About the no-answer text: maybe your version is a bit long :sweat_smile: How about only "Haystack found no good answer to your question. Are you sure this information can be found on Wikipedia?"

Reply:
Ok let's shorten it. However, we should not make this wikipedia-specific as users still use the same stack for demos on their own dataset.
 Maybe:
"Low confidence in answering this question. Treat answers below with caution and try to reformulate your query!"

Reply:
I like the message behind "Low confidence in answering this question." and "Treat answers below with caution." but "confidence" as a term is more technical than what we need in a demo.
Further, reformulation of the query is one thing to try but it could well be that the info is not contained in the indexed documents. I think we need to mention both.

An idea: "Haystack is unsure whether any of the documents contain an answer to your question. Try to reformulate it!"

Reply:
I like this one! :+1:

Reply:
Ok then let's stay pragmatic and go with this one... if we see that people get confused when this message is shown somewhere in the middle of answers, we can iterate on it or filter it out